### PR TITLE
Feat: Add 12 Dev nodes/cleanup workflow files

### LINKED ISSUE #855: Scale the system to 10+ nodes
### Description
Ideally, we should have 12 nodes in a month.

### LINKED ISSUE #868: Organize and improve GitHub acitons
### Description

- Assign proper names
- Remove unnecessary configuration
- Add dev env deployment action

### PR DESCRIPTION
**Pull request by kmaus-near**
A few things in this pull request, but should close #855 at least for Dev. Not sure if we will be running additional nodes ourselves or onboarding more partners, so we can leave that open if need be.

I also cleaned up and changed names of some of the Workflow files which would close #868. There is a roadmap for GitHub to release the ability to organize Workflows into folders, but it does not seem to be on their priority list for this year, so I think this is the best I can do as far as cleanup. I also set the Dev node deployment to be triggered manually, and it will restart the nodes each time, even if there are no changes. Let me know if there are any issues with what I've got.

### CODE REVIEW

File: infra/modules/mig_template/main.tf [hunk 1]
```diff
@@ -156,6 +156,7 @@ resource "google_compute_instance_template" "tpl" {
 
   lifecycle {
     create_before_destroy = "true"
+    ignore_changes = [ disk[0].source_image, labels ]
```
Comment (volovyks) [inline infra/modules/mig_template/main.tf:159]: What does this line mean?

File: .github/workflows/multichain-contract.yml [hunk 2]
```diff
@@ -1,42 +0,0 @@
-name: Multichain Contract
```
Comment (volovyks) [inline .github/workflows/multichain-contract.yml:1]: How do we deploy the contract now?

**Review by volovyks — APPROVED** [review]
LGTM
@ppca please, take a look

File: infra/modules/mig_template/main.tf (hunk 1 above)
Comment (ppca) [reply infra/modules/mig_template/main.tf:159]: I think this likely ignores whether image changes or not, whenever you push, the nodes will be updated.

**Review by ppca — APPROVED** [review]

File: infra/modules/mig_template/main.tf (hunk 1 above)
Comment (kmaus-near) [reply infra/modules/mig_template/main.tf:159]: Correct, this was to stop any unnecessary redeployments when adding nodes.

File: .github/workflows/multichain-contract.yml (hunk 2 above)
Comment (kmaus-near) [reply .github/workflows/multichain-contract.yml:1]: By using [deploy-multichain-dev-contract.yml](https://github.com/near/mpc/pull/872/files#diff-c836f17af97ffd34fd2582c700d283714a220669cd924785e34acc15255d2fe3), I don't think this workflow file is in use.

**Pull request merged**